tools/raisinbread_refresh.php

```diff
@@ -0,0 +1,358 @@
+    'Testing connection to database. No data will be affected at this point.'
+);
+try {
+    // Require generic libraries. In some cases, this can fail if this script
+    // has already dropped tables as the generic_includes.php requires a
+    // database connection.
+    include_once 'generic_includes.php';
+
+    $urlConfigSetting  = $config->getSetting('url');
+    $baseConfigSetting = $config->getSetting('base');
+    $hostConfigSetting = $config->getSetting('host');
+} catch (\DatabaseException $e) {
+    printWarning(
+        "Could not connect to the database in the Config file. " .
+        "It's possible that the LORIS tables have already been dropped."
+    );
+    echo 'Continue attempting to install Raisinbread database? (y/N)' . PHP_EOL;
+    $input = trim(fgets(STDIN));
+
+    if (mb_strtolower($input) !== 'y') {
+        die;
+    }
+}
+
+// Get database information from project's configuration file.
+$config   = $config ?? \NDB_Factory::singleton()->config();
+$dbInfo   = $config->getSettingFromXML('database');
+$dbname   = $dbInfo['database'];
+$host     = $dbInfo['host'];
+$username = $dbInfo['adminUser'];
+$password = $dbInfo['adminPassword'];
+
+if (! $config->getSetting('dev')['sandbox']) {
+    fwrite(
+        STDERR,
+        "Config file indicates that this is not a sandbox. Aborting to " .
+        "prevent accidental data loss." . PHP_EOL
+    );
+    exit(1);
+}
+
+echo <<<CONFIRMATION
+Please type the name of your database `$dbname` to confirm you wish to drop
+tables and import test data: 
+CONFIRMATION;
+
+$input = trim(fgets(STDIN));
+if ($input !== $dbname) {
+    die(printWarning('Input did not match database name. Exiting.'));
+}
+
+$mysqlCommand = <<<CMD
+mysql -A $dbname
+CMD;
+
+echo 'Checking connection via MySQL configuration file...' . PHP_EOL;
+// Test whether a connection to MySQL is possible via a MySQL config file.
+exec($mysqlCommand . ' -e "show tables;" 2>&1 1>/dev/null', $output, $status);
+if ($status != 0) {
+    echo 'Checking connection via project configuration file...' . PHP_EOL;
+    // If not, read DB information from the project's config file. This method
+    // is not as preferable because it generates MySQL warnings due to the
+    // password being supplied via a command-line argument
+
+    // If any of the following variables are null or empty strings, the script
+    // must exit because a connection to the database is impossible.
+    if (empty($host)
+        || empty($dbname)
+        || empty($username)
+        || empty($password)
+    ) {
+        die(
+            printWarning(
+                "Could not connect to database. One of the settings: "
+                . "`host`, `database`, `username` or `password` "
+                . " are missing from your project's configuration file."
+            )
+        );
+    }
+
+    // Try connecting by supplying parameters on command line.
+    $mysqlCommand = <<<CMD
+mysql -A "$dbname" -u "$username" -h "$host" -p$password
```

Comment:
Also not really a reasonable use of the style of string and not properly escaping the arguments.